include/nuttx/mutex.h

```diff
@@ -36,10 +36,11 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-#define NXMUTEX_NO_HOLDER      ((pid_t)-1)
-#define NXMUTEX_INITIALIZER    {NXSEM_INITIALIZER(1, SEM_TYPE_MUTEX | \
-                                SEM_PRIO_INHERIT), NXMUTEX_NO_HOLDER}
-#define NXRMUTEX_INITIALIZER   {NXMUTEX_INITIALIZER, 0}
+#define NXMUTEX_INITIALIZER  {                                             \
+  NXSEM_INITIALIZER(NXMUTEX_NO_HOLDER, SEM_TYPE_MUTEX | SEM_PRIO_INHERIT), \
+  NXMUTEX_NO_HOLDER}
```

Comment:
why not remove holder at line 52?

Reply:
I wanted to left the cleanup of nxmutex for a next PR, this is already a big one.

Reply:
can we add the cleanup patch into this pr?

Reply:
it's fine if the new change hold in a new patch, I want to see the final result which is more clean and understandable.